libsqlite3-sys/build.rs

```diff
@@ -593,47 +712,377 @@ mod bindings {
+        wrapper_fn_inputs_map.insert(
+            api_fn_name.to_string(),
+            generate_varargs_input_idents(field_ident, &bare_fn, &[]),
+        );
+
+        // handle variadic api functions by generating additional bindings for specific sets of method arguments that we support
+        if bare_fn.variadic.is_some() {
+            let const_c_char_type: syn::Type = syn::parse2(quote!(*const ::std::os::raw::c_char))
+                .expect("failed to parse c_char type");
+            let mut_void_type: syn::Type =
+                syn::parse2(quote!(*mut ::core::ffi::c_void)).expect("failed to parse c_char type");
+            let c_int_type: syn::Type =
+                syn::parse2(quote!(::std::os::raw::c_int)).expect("failed to parse c_int type");
+            let mut_c_int_type: syn::Type = syn::parse2(quote!(*mut ::std::os::raw::c_int))
+                .expect("failed to parse mutable c_int reference");
+            // until rust c_variadic support exists, we can't
+            // transparently wrap variadic api functions.
+            // generate specific set of args in place of
+            // variadic for each function we care about.
+            match api_fn_name {
+                "sqlite3_db_config" => {
+                    // https://sqlite.org/c3ref/c_dbconfig_defensive.html
+                    wrapper_fn_inputs_map.insert(
+                        "sqlite3_db_config_constchar".to_string(),
+                        generate_varargs_input_idents(field_ident, &bare_fn, &[&const_c_char_type]),
+                    ); // used for SQLITE_DBCONFIG_MAINDBNAME
+                    wrapper_fn_inputs_map.insert(
+                        "sqlite3_db_config_void_int_mutint".to_string(),
+                        generate_varargs_input_idents(
+                            field_ident,
+                            &bare_fn,
+                            &[&mut_void_type, &c_int_type, &mut_c_int_type],
+                        ),
+                    ); // used for SQLITE_DBCONFIG_LOOKASIDE
+                    wrapper_fn_inputs_map.insert(
+                        "sqlite3_db_config_int_mutint".to_string(),
+                        generate_varargs_input_idents(
+                            field_ident,
+                            &bare_fn,
+                            &[&c_int_type, &mut_c_int_type],
+                        ),
+                    ); // used for all other configuration verbs
+                }
+                "sqlite3_vtab_config" => {
+                    // https://sqlite.org/c3ref/c_vtab_constraint_support.html
+                    wrapper_fn_inputs_map.insert(
+                        "sqlite3_vtab_config_int".to_string(),
+                        generate_varargs_input_idents(field_ident, &bare_fn, &[&c_int_type]),
+                    ); // used for SQLITE_VTAB_CONSTRAINT_SUPPORT
+                }
+                _ => {}
+            };
+        }
```

Comment:
An older version of this code was commented on by @thomcc in https://github.com/rusqlite/rusqlite/pull/631/files/cf658af142399be749bacb0387d54ce836b85f1a#r404006799
"
This isn't really safe. Vararg calling convention isn't guaranteed to be the same as the fixed arg calling convention on all platforms / compilers (https://www.gnu.org/software/libc/manual/html_node/Calling-Variadics.html mentions this).

That said I guess it should be fine in practice, but still is worrisome...
"

Reply:
Don't currently have a proposal to address this. At the moment we are only implementing calls to a small number of variadic API functions (that I am actually using). 

@thomcc would you be happier if inclusion of these bindings was behind a feature? Something like `experimental_variadic_support`?

Reply:
Hmm, I'm not sure if I wrote this elsewhere — worth noting that aarch64 darwin (so ios/tvos/macos m1) is a target where the varargs and fixargs calling conventions are different, so it's not really a theoretical concern.

That said I need to look over the code again, I'm unsure why it can't work as varargs.